src/Native/System.Security.Cryptography.Native/openssl.c

```diff
@@ -1063,6 +1063,25 @@ LockingCallback(int mode, int n, const char* file, int line)
     }
 }
 
+#if __APPLE__
+/*
+Function:
+GetCurrentThreadId
+
+Called back by OpenSSL to get the current thread id.
+
+This is necessary because OSX uses an earlier version of
+OpenSSL, which requires setting the CRYPTO_set_id_callback.
+*/
+unsigned long
```

Comment:
I'm not a C guru, but since we don't really need to target this function outside of this file, can/should it be declared static?